pkgs/development/tools/global-platform-pro/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv, fetchFromGitHub, jdk, maven, writeText, makeWrapper, jre_headless, pcsclite }:
+
+# TODO: This is quite a bit of duplicated logic with gephy. Factor it out?
+stdenv.mkDerivation rec {
+    pname = "global-platform-pro";
+    version = "0.3.10-rc11";
+    describeVersion = "v0.3.10-rc11-0-g8923747"; # git describe --tags --always --long --dirty
+    name = "${pname}-${version}";
+
+    src = fetchFromGitHub {
+        owner = "martinpaljak";
+        repo = "GlobalPlatformPro";
+        rev = "v${version}";
+        sha256 = "0rk81x2y7vx1caxm6wa59fjrfxmjn7s8yxaxm764p8m2qxk3m4y2";
+    };
+
+    patches = [ (writeText "${name}-version.patch" ''
+        diff --git a/pom.xml b/pom.xml
+        index 1e5a82d..1aa01fe 100644
+        --- a/pom.xml
+        +++ b/pom.xml
+        @@ -121,14 +121,10 @@
+                             </execution>
+                         </executions>
+                         <configuration>
+        -                    <executable>git</executable>
+        +                    <executable>echo</executable>
+                             <outputFile>target/generated-resources/pro/javacard/gp/pro_version.txt</outputFile>
+                             <arguments>
+        -                        <argument>describe</argument>
+        -                        <argument>--tags</argument>
+        -                        <argument>--always</argument>
+        -                        <argument>--long</argument>
+        -                        <argument>--dirty</argument>
+        +                        <argument>${describeVersion}</argument>
+                             </arguments>
+                         </configuration>
+                     </plugin>
+    '') ];
+
+    deps = stdenv.mkDerivation {
+        name = "${name}-deps";
+        inherit src patches;
+        nativeBuildInputs = [ jdk maven ];
+        buildPhase = ''
+            while mvn package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
```

Comment:
- please quote the shell variable
- isn't this equivalent to `while ! mvn package ...`?
- is it possible to end up in an endless loop?
- what does that actually do? Download shouldn't be possible in the buildPhase

Reply:
This is part of the code copy/pasted and adapted from the `gephy` package. Basically copied it and twisted it until it compiled, I don't really understand maven :/

That said for your questions:
 * Done
 * I… think it is. Tried building with `while ! mvn package …` and it appears to build, at least :)
 * AFAIK it just re-attempts downloading the dependencies every time the download fails. I guess the author of the `gephy` derivation put it there due to unreliability of the maven repository, and so let it here because it's hard to know whether I'm just having luck downloading all at once or not :)
 * It can download thanks to it being a fixed-output derivation :) (the `outputHash*` variables)

Reply:
- I… think it is. Tried building with while ! mvn package … and it appears to build, at least :)

Then I think using that would be a bit nicer :) Although it is not 100% equivalent since `! command` will be true at any exit code other than 0, I think that is what is meant anyways.

Reply:
I don't find any mention of `outputHash`, what do I miss?

It would probably be valuable to get the author of gephy's expression here.

Reply:
It's the three lines at https://github.com/NixOS/nixpkgs/pull/44288/files/cccfa8f3953933147757d6790ad025e88de32dc0#diff-3f80555b27d7cb249ab01d104b052e6eR52 :)

For the equivalence, I'd guess maybe `mvn package` returns a different error code depending on whether it's a download failure or some other kind of failure? cc @Radvendii

Reply:
And why is it run in a while loop in the first place @Radvendii?